packages/kbn-pm/src/commands/watch.ts

```diff
@@ -0,0 +1,110 @@
+import chalk from 'chalk';
+import { Observable, Subject } from 'rxjs';
+
+import { Project } from '../utils/project';
+import { topologicallyBatchProjects } from '../utils/projects';
+import { parallelizeBatches } from '../utils/parallelize';
+import { Command } from './';
+
+/**
+ * Name of the script in the package/project package.json file to run during `kbn watch`.
+ */
+const watchScriptName = 'kbn:watch';
+
+/**
+ * Number of milliseconds we wait before we fall back to the default watch handler.
+ */
+const defaultHandlerDelay = 3000;
+
+/**
+ * If default watch handler is used, then it's the number of milliseconds we wait for
+ * any build output before we consider watch task ready.
+ */
+const defaultHandlerReadinessTimeout = 2000;
+
+function getWatchHandlers(buildOutput$: Observable<string>) {
+  const typescriptHandler = buildOutput$
+    .first(data => data.includes('$ tsc'))
+    .map(() =>
+      buildOutput$
+        .first(data => data.includes('Compilation complete.'))
+        .mapTo('tsc')
+    );
+
+  const webpackHandler = buildOutput$
+    .first(data => data.includes('$ webpack'))
+    .map(() =>
+      buildOutput$.first(data => data.includes('Chunk Names')).mapTo('webpack')
+    );
+
+  const defaultHandler = Observable.of(void 0)
+    .delay(defaultHandlerReadinessTimeout)
+    .map(() =>
+      buildOutput$
+        .timeout(defaultHandlerDelay)
+        .catch(() => Observable.of(void 0))
+        .mapTo('timeout')
+    );
+
+  return [typescriptHandler, webpackHandler, defaultHandler];
+}
+
+function runScriptUntilWatchIsReady(project: Project) {
+  const buildOutput$ = new Subject<string>();
+  const onDataListener = (data: Buffer) =>
+    buildOutput$.next(data.toString('utf-8'));
+  const onEndListener = () => buildOutput$.complete();
+  const onErrorListener = (e: Error) => buildOutput$.error(e);
+
+  const stream = project.runScriptStreaming(watchScriptName);
```

Comment:
__note:__ I have mixed feelings about reusing this method: it's unusual to see `kibana: ` tagged console output, but from the other side we run several scripts in parallel and without tagging it may look like a mess.

Reply:
I think it's okey, especially as we have the other outputs. We could potentially decide to add a `{ prefix: false }` type param to `runScriptStreaming` that we only called when running `kibana`, which would give Kibana no `kibana:` prefix but all others would have it. But that's stuff we can explore later, as we get more of a feel for what feels right when we start using it actively.